modules/Todo.js

```diff
@@ -0,0 +1,100 @@
+    this.index = this.todos.length;
+    if (localStorage.getItem('todos')) {
+      this.todos = JSON.parse(localStorage.getItem('todos'));
+    }
+  }
+
+  addTodo(description, completed, index) {
+    const newTodo = { description, completed, index };
+    this.todos.push(newTodo);
+
+    localStorage.setItem('todos', JSON.stringify(this.todos));
+    this.displayTodo();
+  }
+
+  removeTodo(index) {
+    this.todos.splice(index, 1);
+
+    localStorage.setItem('todos', JSON.stringify(this.todos));
+
+    this.displayTodo();
+  }
+
+  displayTodo() {
+    const todos = document.querySelector('.todos');
+    todos.innerHTML = '';
+
+    this.todos.forEach((todo, index) => {
+      const todoItem = document.createElement('div');
+      todoItem.className = 'todo-item';
+
+      const todoCheckerButton = document.createElement('img');
+      todoCheckerButton.setAttribute('src', `${Checkbox}`);
+      todoCheckerButton.setAttribute('alt', 'checkbox');
+
+      const todoDescription = document.createElement('input');
+      todoDescription.className = 'todo-list';
+      todoDescription.setAttribute('readonly', 'readonly');
+      todoDescription.type = 'text';
+      todoDescription.className = 'todo-description';
+      todoDescription.value = `${todo.description}`;
+
+      const menuDiv = document.createElement('div');
+      menuDiv.className = 'menu';
+
+      const todoVerticalMenu = document.createElement('img');
+      todoVerticalMenu.setAttribute('src', `${VerticalMenu}`);
+      todoVerticalMenu.setAttribute('alt', 'vertical-menu');
+      todoVerticalMenu.className = 'vertical-menu';
+
+      const todoTrash = document.createElement('img');
+      todoTrash.setAttribute('src', `${TrashCan}`);
+      todoTrash.setAttribute('alt', 'todo-trash');
+      todoTrash.className = 'todo-trash';
+
+      menuDiv.appendChild(todoTrash);
+      menuDiv.appendChild(todoVerticalMenu);
+
+      todoItem.appendChild(todoCheckerButton);
+      todoItem.appendChild(todoDescription);
+      todoItem.appendChild(menuDiv);
+
+      todoItem.addEventListener('mouseenter', (e) => {
+        e.preventDefault();
+        todoItem.classList.add('active');
+        todoDescription.classList.add('active');
+        todoVerticalMenu.classList.add('active');
+        todoTrash.classList.add('active');
+
+        todoDescription.removeAttribute('readonly');
+        todoDescription.focus();
+
+        todoTrash.addEventListener('click', (e) => {
+          e.preventDefault();
+          this.removeTodo(index);
+        });
+      });
+
+      todoItem.addEventListener('mouseleave', (e) => {
+        e.preventDefault();
+        todoItem.classList.remove('active');
+        todoDescription.classList.remove('active');
+        todoVerticalMenu.classList.remove('active');
+        todoTrash.classList.remove('active');
+
+        todoDescription.setAttribute('readonly', 'readonly');
+      });
+
+      todos.appendChild(todoItem);
+    });
+  }
```

Comment:
- [ ] [REQUIRED] Hello  Prince, you have done a great job here👍🏽 🥇 
However, a part of the project requirement states that you should ensure by default, new tasks should have the property completed set to false. Let's implement that, please.

- [ ] [REQUIRED] Also, kindly ensure that the new tasks have the property `index` set to the value of the new array length (i.e. if a 5th task is added to the list, the index of that task should equal to 5) and let's also take note that ...

- [ ] [REQUIRED] Deleting a task should update all remaining items' indexes, so they represent the current list order and are unique.

### Your Design
<img width="706" alt="Screenshot 2023-03-30 at 00 34 53" src="https://user-images.githubusercontent.com/58692590/228691921-23f2467b-bb2e-448b-a622-58965ddbdf85.png">

- What you currently have is just the `description` key && value, let's implement that of the `index` and the `completed` as well.
Let's fix these 💪🏽